### PR TITLE
stylelint integration into the build

### PR DESCRIPTION
**Pull request by reblace**
Integrating style (scss) linting into the build.

- Using `stylelint` along with most default options for scss linting
- Made some minor changes to style files to fix some lint issues
- Disabled a couple of the lint rules to accommodate some harder to follow rules
- Added linting as part of the `build` script for npm

One of the bigger changes to styles I made was to remove the `alert` selector from `system-alert.component.scss` because `alert` is not a valid html tag. Since that selector just wrapped the whole scss style, I can't imagine it's going to change anything to have removed it. Especially considering that styles are scoped.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,12 +5,13 @@
     "ng": "ng",
     "start": "ng serve --proxy-config proxy.conf.json",
     "start:prod": "ng serve --proxy-config proxy.conf.json --prod",
-    "build": "ng build",
-    "build:prod": "ng build --prod",
+    "build": "npm run lint && ng build",
```
Comment (brianghig) [inline package.json:8]: @reblace thanks for including these lint tasks with each build. Should we add these details to the top-level README under the NPM Scripts section? https://github.com/asymmetrik/ngx-starter#npm-scripts

**Review by brianghig — CHANGES_REQUESTED** [review]
README docs look like the only thing that's missing for this update. Thanks for incorporating another automated code analysis tool!

File: package.json (hunk 1 above)
Comment (reblace) [reply package.json:8]: Yeah, do we want linting to happen with the build? Or, do we only want it as part of a ci build?

Comment (reblace) [reply package.json:8]: I collapsed both linting operations into the single lint npm script. We'll leave running the lint as part of the build for now.

**Review by brianghig — APPROVED** [review]

**Pull request merged**